addOns/pscanrulesAlpha/src/main/java/org/zaproxy/zap/extension/pscanrulesAlpha/SubResourceIntegrityAttributeScanRule.java

```diff
@@ -114,9 +125,66 @@ public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
                                         .setCweId(345) // CWE-345: Insufficient Verification of Data
                                         // Authenticity
                                         .setWascId(15) // Application Misconfiguration
+                                        .setOtherInfo(getOtherInfo(msg, element, tree))
                                         .raise());
     }
 
+    private String calculateIntegrityHash(HttpMessage msg, Element element, SiteMap tree) {
+        String src = element.getAttributeValue("src");
+        String integrityHash = "";
+        if (src == null) {
+            return "";
+        }
+        // if (msg.isResponseFromTargetHost()) {
+        try {
+            URI newURI = new URI(msg.getRequestHeader().getURI().toString());
+            newURI = newURI.resolve(src);
+            SiteNode node =
+                    tree.findNode(new org.apache.commons.httpclient.URI(newURI.toString(), true));
+            integrityHash =
+                    Base64.getEncoder()
+                            .encodeToString(
+                                    DigestUtils.sha384Hex(
+                                                    node.getHistoryReference()
+                                                            .getHttpMessage()
+                                                            .getResponseBody()
+                                                            .toString())
+                                            .getBytes());
+        } catch (Exception e) {
+            logger.debug("Error occured while calculating the hash. Error: {}", e.getMessage(), e);
+        }
+        // }
+        return integrityHash;
+    }
+
+    private String getOtherInfo(HttpMessage msg, Element element, SiteMap tree) {
+        String integrityHash = calculateIntegrityHash(msg, element, tree);
+        String otherInfo = "";
+        if (integrityHash.isEmpty()) {
+            return otherInfo;
+        }
+        if (checkIntegrity(integrityHash, element)) {
+            otherInfo = Constant.messages.getString(MESSAGE_PREFIX + "otherinfo", integrityHash);
+            return otherInfo;
+        } else {
+            otherInfo =
+                    "Calculated hash does not match provided hash. Hash Calculated: "
+                            + integrityHash;
+            return otherInfo;
+        }
+    }
+
+    private boolean checkIntegrity(String integrityHash, Element element) {
+        String srcintegrity = element.getAttributeValue("integrity");
+        if ((srcintegrity != null) && (integrityHash.equals(srcintegrity))) {
```

Comment:
Could remove the inner brackets

Reply:
The right hand check will be false always, the integrity value has the algorithm as prefix (which is not being added yet), also it might be using other algorithm than the one used by the code.

Reply:
The previous comment was not addressed.

Reply:
I've renamed the method to `isIntegrityMissing` and turned the if/then/else into a single return `return srcIntegrity != null && !integrityHash.isEmpty();`

Reply:
I changed the method to `isIntegrityMissing` and the if/then/else to a single return `return srcIntegrity == null || integrityHash.isEmpty();`

Reply:
That does not address the comment, now is not even checking the value and just assuming it's different always.

Reply:
I guess I don't understand the comment then :(

Reply:
Before checking if they are equal it needs to check if both are using the same algorithm.

Reply:
Okay, now I get what you're suggesting.

Reply:
Do we really need to handle the situation where the present hash (if it happens to be sha384) and doesn't match.

I agree that it might be kinda handy but it's turning into a major overhaul of the existing functionality to make it work.

Reply:
I'm ok if that's not checked.

Reply:
Perfect, thanks.